base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,31 +21,248 @@
+    HWND hCombo = GetDlgItem(hwnd, IDC_NOTIFICATION_BEHAVIOUR);
+
+    if (ComboBox_GetCount(hCombo) == 0)
+    {
+        WCHAR sShow[100];
+        WCHAR sHide[100];
+        WCHAR sInactive[100];
+        LoadStringW(NULL, IDS_NOTIF_BEH_SHOW, sShow, _countof(sShow));
+        LoadStringW(NULL, IDS_NOTIF_BEH_HIDE, sHide, _countof(sHide));
+        LoadStringW(NULL, IDS_NOTIF_BEH_HIDE_INACTIVE, sInactive, _countof(sInactive));
+        ComboBox_AddString(hCombo, sShow);
+        ComboBox_AddString(hCombo, sHide);
+        ComboBox_AddString(hCombo, sInactive);
+        ComboBox_SetCurSel(hCombo, 0);
+    }
+
+    int iItem = itemdata->iItem;
+    if (iItem > -1)
+    {
+        POINT pt;
+        RECT rc;
+        ListView_GetItemPosition(hListView, iItem, &pt);
+        GetWindowRect(hListView, &rc);
+        POINT lv = { rc.left, rc.top };
+        ScreenToClient(hwnd, &lv);
+
+        LVITEM lvItem;
+        lvItem.iItem = iItem;
+        lvItem.iSubItem = 0;
+        lvItem.mask = LVIF_PARAM;
+        ListView_GetItem(hListView, &lvItem);
+
+        ComboBox_SetCurSel(hCombo, lvItem.lParam);
+
+        ComboBox_SetItemData(hCombo, 0, iItem);
+        SetWindowPos(hCombo, HWND_TOP, 190 + lv.x, pt.y + lv.y, 135, 20, SWP_SHOWWINDOW);
+    }
+    else
+    {
+        ShowWindow(hCombo, SW_HIDE);
+    }
+}
+
+VOID SetItemBehavior(HWND hwnd, HWND hCombo)
+{
+    int iState = ComboBox_GetCurSel(hCombo);
+    int iItem = (int)ComboBox_GetItemData(hCombo, 0);
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+
+    WCHAR sState[100];
+    int resId = iState == 0 ? IDS_NOTIF_BEH_SHOW : (iState == 1 ? IDS_NOTIF_BEH_HIDE : IDS_NOTIF_BEH_HIDE_INACTIVE);
+    LoadStringW(NULL, resId, sState, _countof(sState));
+
+    LVITEM lvItem;
+    ZeroMemory(&lvItem, sizeof(lvItem));
+    lvItem.mask = LVIF_TEXT;
+    lvItem.iItem = iItem;
+    lvItem.iSubItem = 1;
+    lvItem.cchTextMax = 100;
+    lvItem.pszText = sState;
+    ListView_SetItem(hListView, &lvItem);
+
+    LVITEM lvItem2;
+    ZeroMemory(&lvItem2, sizeof(lvItem2));
+    lvItem2.mask = LVIF_PARAM;
+    lvItem2.iSubItem = 0;
+    lvItem2.lParam = iState;
+    lvItem2.iItem = iItem;
+    ListView_SetItem(hListView, &lvItem2);
+
+    ShowWindow(hCombo, SW_HIDE);
+}
+
+VOID ResetBehaviors(HWND hwnd)
+{
+    HWND hListView = GetDlgItem(hwnd, IDC_NOTIFICATION_LIST);
+    int count = ListView_GetItemCount(hListView);
+    WCHAR state[100]; 
+    LoadStringW(NULL, IDS_NOTIF_BEH_HIDE_INACTIVE, state, _countof(state));
+
+    LVITEM lvItem;
+    ZeroMemory(&lvItem, sizeof(lvItem));
+    lvItem.mask = LVIF_TEXT;
+    lvItem.iSubItem = 1;
+    lvItem.cchTextMax = 100;
+    lvItem.pszText = state;
+
+    LVITEM lvItem2;
+    ZeroMemory(&lvItem2, sizeof(lvItem));
+    lvItem2.mask = LVIF_PARAM;
+    lvItem2.iSubItem = 0;
+    lvItem2.lParam = 2;
+
+    for(int i = 0; i< count; i++)
+    {
+        lvItem2.iItem = i;
+        ListView_SetItem(hListView, &lvItem2);
```

Comment:
Why using two different lvItems?

Reply:
no need to whole reinitialize them on every iteration